auth-api/migrations/versions/304705f06971_tasks_multiple_reasons.py

```diff
@@ -0,0 +1,84 @@
+"""tasks_multiple_reasons
+
+Revision ID: 304705f06971
+Revises: 41ac4080a7ed
+Create Date: 2021-09-10 14:50:36.964423
+
+"""
+from typing import List
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+from sqlalchemy import text
+
+from auth_api.models import Task
+
+# revision identifiers, used by Alembic.
+revision = '304705f06971'
+down_revision = '41ac4080a7ed'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # We get the list of tasks at first.
+    # we convert tasks.remarks from character type to array form ( check if there are arrays already, edge case )
+    # we convert the column remarks type from character to ARRAY
+
+    conn = op.get_bind()
+    task_res = conn.execute("SELECT * FROM tasks;")
+    task_list: List[Task] = task_res.fetchall()
+
+    for task in task_list:
+        if task.remarks is None:
+            continue
+        # A check if there are task remarks that is already wrapped in '{}', ideally, we would not hit this scenario
+        if task.remarks.startswith('{') and task.remarks.endswith('}'):
+            continue
+        else:
+            task_id = task.id
+            remarks = task.remarks.split(',')
+            # Now updating the new remarks value back in the table
+            update_sql = text("UPDATE tasks SET remarks = :remarks where id = :task_id") \
+                .params(
+                remarks=remarks, task_id=task_id)
+            op.execute(update_sql)
+
+    # ### alter command auto generated by Alembic ###
+    op.alter_column('tasks', 'remarks',
```

Comment:
shudnt this happen before the update statement?

Reply:
ARRAY type expects JSON value and would not accept string literal.